### DISCUSSION TITLE
useEffect, useState, onclick not working

### DISCUSSION THREAD

**Discussion by wibed** [module-federation/module-federation-examples#3116] [Q&A]
using react-18-ssr https://github.com/module-federation/module-federation-examples/tree/master/react-18-ssr

example i am stuck at using client side element as named in the title.
anyone experiencing the same issue?

**Comment by XavierLeTohic** [comment]
Makes sure you included the client bundle `main.js` as bootstrap script like in the [render.tsx](https://github.com/module-federation/module-federation-examples/blob/master/react-18-ssr/shell/src/server/render.tsx#L27C64-L27C64) file of the example.

Alternatively you can add it to `bootstrapScripts` of the options of `renderToPipeableStream`

```tsx
  const stream = renderToPipeableStream(<App />, {
    bootstrapScripts: ['/static/main.js'],
    onAllReady() {
      // ...
    }
```

As mentioned in the [React documentation](https://react.dev/reference/react-dom/server/renderToPipeableStream#parameters) if you omit it it won't hydrate your components at all.
